File: src/main/java/de/akquinet/ccsp/floodwatch/datamodel/Measurement.java
package de.akquinet.ccsp.floodwatch.datamodel;

import java.util.Date;
import lombok.Data;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data
public class Measurement {

    private String id;

    @DBRef
    private final MeasurementImport measurementImport;

    @DBRef
    private final Location location;

    private final Date timeOfMeasurement;

    private int waterLevelInCm;

    private String waterLevelGroup;

    private float throughFlowInCubicMetersPerSecond;

    private String throughFlowGroup;

    private float waterTemperatureInCelsius;
}

File: src/main/kotlin/de/akquinet/ccsp/floodwatch/datamodel/Measurement.kt
package de.akquinet.ccsp.floodwatch.datamodel

import org.springframework.data.mongodb.core.mapping.DBRef
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@Document
data class Measurement(@field:DBRef val measurementImport: MeasurementImport,
                       @field:DBRef val location: Location,
                       val timeOfMeasurement: Date) {
    var id: String? = null
    var waterLevelInCm: Int? = null
    var waterLevelGroup: String? = null
    var throughFlowInCubicMetersPerSecond: Float? = null
    var throughFlowGroup: String? = null
    var waterTemperatureInCelsius: Float? = null
}
